Sources/Services/ContainerImagesService/Server/SnapshotStore.swift

```diff
@@ -21,21 +21,41 @@ import ContainerizationExtras
 import ContainerizationOCI
 import ContainerizationOS
 import Foundation
+import Logging
 import TerminalProgress
 
 public actor SnapshotStore {
     private static let snapshotFileName = "snapshot"
     private static let snapshotInfoFileName = "snapshot-info"
     private static let ingestDirName = "ingest"
 
+    /// Return the Unpacker to use for a given image.
+    /// If the given platform for the image cannot be unpacked return `nil`.
+    public typealias UnpackStrategy = @Sendable (Containerization.Image, Platform) async throws -> Unpacker?
+
+    public static let defaultUnpackStrategy: UnpackStrategy = { image, platform in
+        guard platform.os == "linux" else {
+            return nil
+        }
+        var minBlockSize = 512.gib()
+        if image.reference == ClientDefaults.get(key: .defaultInitImage) {
+            minBlockSize = 512.mib()
```

Comment:
Let's see if we can move this to a UserDefault in a followup so that end users can customize this size.